src/main/java/com/aws/greengrass/util/platforms/unix/UnixPlatform.java

```diff
@@ -481,45 +481,37 @@ protected void runCmd(String cmdStr, Consumer<CharSequence> out, String msg)
         }
     }
 
-    Set<Integer> getChildPids(Process process) throws IOException, InterruptedException {
-        PidProcess pp = Processes.newPidProcess(process);
-
-        // Use PS to list process PID and parent PID so that we can identify the process tree
-        logger.atDebug().log("Running ps to identify child processes of pid {}", pp.getPid());
-        Process proc = Runtime.getRuntime().exec(new String[]{"ps", "-ax", "-o", "pid,ppid"});
-        proc.waitFor();
-        if (proc.exitValue() != 0) {
-            logger.atWarn().kv("pid", pp.getPid()).kv("exit-code", proc.exitValue())
-                    .kv(STDOUT, inputStreamToString(proc.getInputStream()))
-                    .kv(STDERR, inputStreamToString(proc.getErrorStream())).log("ps exited non-zero");
-            throw new IOException("ps exited with " + proc.exitValue());
-        }
-
-        try (InputStreamReader reader = new InputStreamReader(proc.getInputStream(), StandardCharsets.UTF_8);
-             BufferedReader br = new BufferedReader(reader)) {
-            Stream<String> lines = br.lines();
-            Map<String, String> pidToParent = lines.map(s -> {
-                Matcher matches = PS_PID_PATTERN.matcher(s.trim());
-                if (matches.matches()) {
-                    return new Pair<>(matches.group(1), matches.group(2));
+    Set<Integer> getChildPids(PidProcess pp) throws IOException, InterruptedException {
+        logger.atDebug().log("Identifying child processes of pid {}", pp.getPid());
+
+        Set<Integer> childPids = new HashSet<>();
+        // Find all tasks (threads) of the process
+        try (Stream<File> tasks = Files.list(Paths.get("/proc", String.valueOf(pp.getPid()), "task"))
+                .map(Path::toFile)
+                .filter(File::isDirectory)) {
+            // For each task, find its children
+            for (File task : tasks.collect(Collectors.toSet())) {
+                String children = new String(Files.readAllBytes(task.toPath().resolve("children")),
+                        StandardCharsets.UTF_8);
+                Set<Integer> childPid =
+                        Arrays.stream(children.split("\\s+"))
+                                .filter(s -> !s.isEmpty())
+                                .map(Integer::parseInt)
+                                .collect(Collectors.toSet());
```

Comment:
I'm doing some reading on this, and this interface is documented as being unreliable because this is meant for CRIU use-cases, specifically: https://man7.org/linux/man-pages/man5/proc.5.html

```
/proc/[pid]/task/[tid]/children (since Linux 3.5)
              A space-separated list of child tasks of this task.  Each
              child task is represented by its TID.

              This option is intended for use by the checkpoint-restore
              (CRIU) system, and reliably provides a list of children
              only if all of the child processes are stopped or frozen.
              It does not work properly if children of the target task
              exit while the file is being read!  Exiting children may
              cause non-exiting children to be omitted from the list.
              This makes this interface even more unreliable than
              classic PID-based approaches if the inspected task and its
              children aren't frozen, and most code should probably not
              use this interface.
```

I'm noticing on my system, the `children` file does not even exist. I suspect we'll need a fallback strategy.